truss/truss_config.py

```diff
@@ -395,6 +395,37 @@ def to_dict(self):
         }
 
 
+@dataclass
+class DockerServer:
+    setup_command: str
+    start_command: str
+    server_port: int
+    readiness_endpoint: str
+    liveness_endpoint: str
+    predict_endpoint: str
+
+    @staticmethod
+    def from_dict(d) -> "DockerServer":
+        return DockerServer(
+            setup_command=d.get("setup_command", ""),
+            start_command=d.get("start_command", ""),
+            server_port=d.get("server_port", 8000),
+            readiness_endpoint=d.get("readiness_endpoint", ""),
+            liveness_endpoint=d.get("liveness_endpoint", ""),
+            predict_endpoint=d.get("predict_endpoint", ""),
+        )
+
+    def to_dict(self):
+        return {
+            "setup_command": self.setup_command,
+            "start_command": self.start_command,
+            "server_port": self.server_port,
+            "readiness_endpoint": self.readiness_endpoint,
+            "liveness_endpoint": self.liveness_endpoint,
+            "predict_endpoint": self.predict_endpoint,
+        }
```

Comment:
@tianshuc0731 you're fine, your just following the pattern of what is here...

@squidarth Why do we write code like this? This seems like missing the point of dataclass/pydantic models? This whole file had me scratch my head multiple times...

Reply:
I'm not sure honestly...I think one of the problems with dataclasses is that it doesn't do any actual data validation (ie: if you pass an int into a string field it'll be happy).

when we address proper truss config validation, we should consider converting all of this to pydantic...

Reply:
This here doesn't validate anything - at all it is source of errors/oversights/divergence and a maintenance drag.